services/venice-router/src/main/java/com/linkedin/venice/router/MetaDataHandler.java

```diff
@@ -670,13 +670,14 @@ private void handleStreamHybridStoreQuotaStatusLookup(ChannelHandlerContext ctx,
       throws IOException {
     String storeName = helper.getResourceName();
     checkResourceName(storeName, "/" + TYPE_STREAM_HYBRID_STORE_QUOTA + "/${storeName}");
-    if (!storeConfigRepo.getStoreConfig(storeName).isPresent()) {
+    Store store = storeRepository.getStore(storeName);
```

Comment:
Will the store config repository be cleaned eventually? Wondering if there is something more fundamental in the way store deletion occurs where the associated state isn't cleaned up consistently?

Reply:
Store config and store repository are different entities.
Store config is one per fabric and it holds cluster discovery and migration information. i.e. current cluster, is store migrating, source cluster and destination cluster.
Store repo is one per fabric per cluster and it holds the store's config and states (versions) in that cluster.
IIUC the scenario Zac is trying to solve here is when the store is deleted from the source cluster because store migration is completed and end migration is called to cleanup the store from source cluster. The store will continue to exist in the store repo of the new cluster and the store config will exist and be updated with the dest cluster.

Reply:
I see. Thanks for the explanation :)